Sources/Datadog/Tracing/Span/SpanBuilder.swift

```diff
@@ -23,50 +23,129 @@ internal struct SpanBuilder {
     /// source tag to encode in span.
     let source: String
 
-    /// Encodes tag `Span` tag values as JSON string
-    private let tagsJSONEncoder: JSONEncoder = .default()
+    func createSpan(
+        traceID: TracingUUID,
+        spanID: TracingUUID,
+        parentSpanID: TracingUUID?,
+        operationName: String,
+        startTime: Date,
+        finishTime: Date,
+        tags: [String: Encodable],
+        baggageItems: [String: String],
+        logFields: [[String: Encodable]]
+    ) -> Span {
```

Comment:
💡 By passing these immutable values we avoid the thread sync penalty when reading them from `DDSpan`. If reading one by one, we'd need `queue.sync {}` on each value. If passing in bulk, we only sync once.